girder_volview/__init__.py

```diff
@@ -1,10 +1,71 @@
+import cherrypy
+import errno
+
 from girder import plugin
 
+from girder.api.describe import Description, autoDescribeRoute
+from girder.api import access 
+from girder.api.rest import boundHandler
+from girder.constants import AccessType, TokenScope
+
+from girder.models.file import File as FileModel
+from girder.models.upload import Upload
+from girder.utility import RequestBodyStream
+from girder.utility.model_importer import ModelImporter
+from girder.exceptions import GirderException
+
+@access.public(cookie=True, scope=TokenScope.DATA_WRITE)
+@boundHandler
+@autoDescribeRoute(
+    Description('Save VolView session in an item')
+    .param('itemId', 'The item ID', paramType='path')
+    .errorResponse())
+def saveVolView(self, itemId):
+    size = int(cherrypy.request.headers.get('Content-Length'))
+
+    # modified from girder.api.v1.file.File.initUpload 
+    fileModel = FileModel()
+    parentId = itemId
+    parentType = 'item'
+    name = 'session.volview.zip'
+    mimeType = 'application/zip'
+    reference = None
+    user = self.getCurrentUser()
+    parent = ModelImporter.model(parentType).load(
```

Comment:
Since we are always using items as the parent, this could avoid using the model importer, which would be more consistent with other girder code: 
```
from girder.models.item import Item
...
parent = Item().load(...)
```